support/src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/BuildTriggerStepExecution.java

```diff
@@ -54,6 +58,7 @@ public boolean start() throws Exception {
         actions.add(new CauseAction(new Cause.UpstreamCause(invokingRun)));
         List<ParameterValue> parameters = step.getParameters();
         if (parameters != null) {
+            completeDefaultParameters(parameters, (Job) project);
```

Comment:
Yes, the issue only comes up when _some but not all_ parameters are set explicitly.
There is already a [test](https://github.com/amuniz/workflow-plugin/blob/JENKINS-31897/aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/build/BuildTriggerStepTest.java#L205) for the _all defaults_ case.